strongbox-web-core/src/main/java/org/carlspring/strongbox/config/ServletConfig.java

```diff
@@ -0,0 +1,75 @@
+package org.carlspring.strongbox.config;
+
+import org.carlspring.strongbox.web.DirectoryTraversalFilter;
+import org.carlspring.strongbox.web.HeaderMappingFilter;
+
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import java.util.EnumSet;
+
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.context.request.RequestContextListener;
+import org.springframework.web.filter.CommonsRequestLoggingFilter;
+import org.springframework.web.filter.DelegatingFilterProxy;
+import org.springframework.web.filter.RequestContextFilter;
+
+@Configuration
+public class ServletConfig
+{
+
+    private static FilterRegistrationBean registerFilter(String filterName,
+                                                         Filter filter)
+    {
+        FilterRegistrationBean registration = new FilterRegistrationBean();
+
+        EnumSet<DispatcherType> dispatcherTypes = getSecurityDispatcherTypes();
+
+        registration.setFilter(filter);
+        registration.setName(filterName);
+        registration.setAsyncSupported(true);
+        registration.setDispatcherTypes(dispatcherTypes);
+        registration.addUrlPatterns("/*");
+
+        return registration;
+    }
+
+    private static EnumSet<DispatcherType> getSecurityDispatcherTypes()
+    {
+        return EnumSet.of(DispatcherType.REQUEST, DispatcherType.ERROR, DispatcherType.ASYNC);
+    }
+
+    @Bean
+    RequestContextListener requestContextListener()
+    {
+        return new RequestContextListener();
+    }
+
+    @Bean
+    FilterRegistrationBean commonsRequestLoggingFilter()
+    {
+        return registerFilter(CommonsRequestLoggingFilter.class.getSimpleName(),
+                              new DelegatingFilterProxy("commonsRequestLoggingFilter"));
+    }
+
+    @Bean
+    FilterRegistrationBean directoryTraversalFilter()
+    {
+        return registerFilter(DirectoryTraversalFilter.class.getSimpleName(), new DirectoryTraversalFilter());
+    }
+
+    @Bean
+    FilterRegistrationBean headerMappingFilter()
```

Comment:
@fuss86 ,

Would you mind explaining this change to me? After this change, on my branch, I'm getting:

```
org.springframework.boot.web.server.WebServerException: Unable to start embedded Jetty server
	at org.springframework.boot.web.embedded.jetty.JettyWebServer.start(JettyWebServer.java:167)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.startWebServer(ServletWebServerApplicationContext.java:300)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.finishRefresh(ServletWebServerApplicationContext.java:162)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:553)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:140)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:395)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:327)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1255)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1243)
	at org.carlspring.strongbox.server.StrongboxSpringBootApplication.main(StrongboxSpringBootApplication.java:36)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.WarLauncher.main(WarLauncher.java:58)
Caused by: org.eclipse.jetty.util.MultiException: Multiple exceptions
	at org.eclipse.jetty.util.MultiException.ifExceptionThrow(MultiException.java:124)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:787)
	at org.springframework.boot.web.embedded.jetty.JettyEmbeddedWebAppContext$JettyEmbeddedServletHandler.deferredInitialize(JettyEmbeddedWebAppContext.java:46)
	at org.springframework.boot.web.embedded.jetty.JettyEmbeddedWebAppContext.deferredInitialize(JettyEmbeddedWebAppContext.java:36)
	at org.springframework.boot.web.embedded.jetty.JettyWebServer.handleDeferredInitialize(JettyWebServer.java:209)
	at org.springframework.boot.web.embedded.jetty.JettyWebServer.start(JettyWebServer.java:141)
	... 18 common frames omitted
	Suppressed: org.springframework.beans.factory.BeanNotOfRequiredTypeException: Bean named 'headerMappingFilter' is expected to be of type 'javax.servlet.Filter' but was actually of type 'org.springframework.boot.web.servlet.FilterRegistrationBean'
		at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:389)
		at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204)
		at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1095)
		at org.springframework.web.filter.DelegatingFilterProxy.initDelegate(DelegatingFilterProxy.java:337)
		at org.springframework.web.filter.DelegatingFilterProxy.initFilterBean(DelegatingFilterProxy.java:242)
		at org.springframework.web.filter.GenericFilterBean.init(GenericFilterBean.java:241)
		at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:139)
		at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:741)
		... 22 common frames omitted
	Suppressed: org.springframework.beans.factory.BeanNotOfRequiredTypeException: Bean named 'commonsRequestLoggingFilter' is expected to be of type 'javax.servlet.Filter' but was actually of type 'org.springframework.boot.web.servlet.FilterRegistrationBean'
		at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:389)
		at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204)
		at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1095)
		at org.springframework.web.filter.DelegatingFilterProxy.initDelegate(DelegatingFilterProxy.java:337)
		at org.springframework.web.filter.DelegatingFilterProxy.initFilterBean(DelegatingFilterProxy.java:242)
		at org.springframework.web.filter.GenericFilterBean.init(GenericFilterBean.java:241)
		at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:139)
		at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:741)
		... 22 common frames omitted
Caused by: org.springframework.beans.factory.BeanNotOfRequiredTypeException: Bean named 'headerMappingFilter' is expected to be of type 'javax.servlet.Filter' but was actually of type 'org.springframework.boot.web.servlet.FilterRegistrationBean'
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:389)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1095)
	at org.springframework.web.filter.DelegatingFilterProxy.initDelegate(DelegatingFilterProxy.java:337)
	at org.springframework.web.filter.DelegatingFilterProxy.initFilterBean(DelegatingFilterProxy.java:242)
	at org.springframework.web.filter.GenericFilterBean.init(GenericFilterBean.java:241)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:139)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:741)
	... 22 common frames omitted
```

Reply:
In my PR, `org.carlspring.strongbox.config.StrongboxWebInitializer` has been removed and filters like headerMappingFilter have been moved to `ServletConfig.java `. I used Spring/Boot default dispatcher configuration.